loopchain/baseservice/websocket_client.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2018 ICON Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Class for websocket client between citizen node and mother peer"""
+
+import asyncio
+import json
+import logging
+
+import websockets
+from websockets.exceptions import InvalidStatusCode
+
+from loopchain import configure as conf
+from loopchain.baseservice import TimerService, Timer
+from loopchain.blockchain import Block
+from loopchain.protos import message_code
+
+
+class WebSocketClient:
+    def __init__(self, channel, peer_target, rs_target, block_height, channel_service):
+        self.__channel = channel
+        self.__peer_target = peer_target
+        self.__block_height = block_height
+        self.__rs_target = rs_target
+        self.__channel_service: 'ChannelService' = channel_service
+        self.__target_uri = f"ws://{self.__rs_target}/api/node/{channel}"
+
+    @property
+    def block_height(self):
+        return self.__block_height
+
+    @block_height.setter
+    def block_height(self, block_height):
+        self.__block_height = block_height
+
+    @property
+    def target_uri(self):
+        return self.__target_uri
+
+    async def subscribe(self, uri):
+        while True:
+            try:
+                async with websockets.connect(uri) as websocket:
+                    await self.__stop_shutdown_timer()
+                    request = json.dumps({
+                        'height': self.__block_height
+                    })
+                    await websocket.send(request)
+
+                    while True:
+                        new_block_json = await websocket.recv()
+                        response_code = await self.__add_confirmed_block(block_json=new_block_json)
+                        if response_code == message_code.Response.success:
```

Comment:
This 'if' statement can be removed, if `self.__add_confimed_block` raise exception.

Reply:
Removed.